dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/internal/JDBCDriverService.java

```diff
@@ -1037,6 +1078,189 @@ protected void setSharedLib(Library lib) {
+        ORACLELOG_FILE_COUNT = "oracleLogFileCount",
+        ORACLELOG_FILENAME = "oracleLogFileName",
+        ORACLELOG_TRACELEVEL = "oracleLogTraceLevel",
+        ORACLELOG_FORMAT = "oracleLogFormat",
+        ORACLELOG_PACKAGENAME = "oracleLogPackageName";
+        
+        Logger parentLogger = Logger.getLogger(ORACLELOG_PARENT_PACKAGENAME);
+        
+        //If the 'oracle' logger isn't using a parent then it's possible the user configured their 
+        //application to send logs somewhere else.  Don't overwrite them.
+        if(!parentLogger.getUseParentHandlers()) {
+            return;
+        }
+        
+        // Fail fast: If some, but not all, required properties were provided tell user and return.
+        if( (System.getProperty(ORACLELOG_FILENAME) != null) == (System.getProperty(ORACLELOG_PACKAGENAME) != null) &&
+            (System.getProperty(ORACLELOG_PACKAGENAME) != null) == Boolean.getBoolean(ORACLELOG_ENABLE_TRACE)) {
+            if(!Boolean.getBoolean(ORACLELOG_ENABLE_TRACE)) { //User enabled logging, but wants logs to go to Liberty
+                return;
+            }
+        } else {
+            if(tc.isInfoEnabled()) {
+                Tr.info(tc, "ORACLE_TRACE_ENABLE_INFO", ORACLELOG_ENABLE_TRACE + ", " + ORACLELOG_FILENAME + ", " + ORACLELOG_PACKAGENAME);
+            }
+            return;
+        }
+        
+        if(tc.isEntryEnabled()) {
+            Tr.entry(tc, method);
+        }
+
+        // Expected Settings from system properties
+        String fileName = null;    //REQUIRED
+        String packageName = null; //REQUIRED
+        int fileSizeLimit = -1;    //Needs validation >= 0
+        int fileCountLimit = -1;   //Needs validation >= 1
+        Formatter formatter;
+        Level traceLevel;
+        
+        // Default values for optional parameters
+        final int defaultFileSizeLimit = 0;  // unlimited
+        final int defaultFileCountLimit = 1; // only one file to rotate through
+        final Formatter defaultFormatter = new SimpleFormatter();
+        final Level defaultLevel = Level.INFO;
+        
+        Map<String, Exception> parseExceptions = new HashMap<>();
+        
+        // Variable Value
+        String holder = null;
+        
+        // Get, parse, and validate system properties
+        holder = System.getProperty(ORACLELOG_FILENAME);
+        if (holder != null && !holder.equals(""))
+            fileName = holder;
+        
+        holder = System.getProperty(ORACLELOG_PACKAGENAME);
+        if (holder != null && !holder.equals(""))
+            packageName = holder;
+        
+        holder = System.getProperty(ORACLELOG_FILE_SIZE_LIMIT);
+        try {
+            if (holder != null && (!holder.equals(""))) {
+                fileSizeLimit = Integer.parseInt(holder); 
+                if(fileSizeLimit < 0) {
+                    throw new NumberFormatException("File size cannot be negative");
```

Comment:
Doesn't this need to be i8n'ed?

Reply:
This exception is meant to be caught and rethrown using the ORACLE_TRACE_PARSE_WARNING=DSRA7043W
But you are correct that I put the exception message in that warning. 

Here are some options: 
1. We could introduce our own translated message here.
2. I could follow the convention of the JDK messages which are not localized `oracleLogFileSizeLimit -1 less than 0`
3. I could have a purely mathematical message: `oracleLogFileSizeLimit < 0`

Thoughts?

Reply:
#3 would be my preference